src/test/PulsarMock.js

```diff
@@ -0,0 +1,251 @@
+export class PulsarClientMock {
+  // A map used to simulate an in-memory store for messages. The keys are topic names, and the values are arrays of message objects.
+  static inMemoryStore = new Map();
```

Comment:
with all these being static, it will not be possible to run more than one test client on a given system. I am not sure that is a good idea ... perhaps good enough for now, but prolly worth the small refactor to make then instance fields

Reply:
not a big, but a feature - I want the data to persist client instance restart, so we can demonstrate how server side acks work